add an env var for path to pre-downloaded flashinfer cubin files

842974287 opened the pull request:
Summary:
Previously https://github.com/vllm-project/vllm/pull/21893 added a check for network access to NV cubin artifact endpoint. And if the check failed, we would opt out from trtllm attn backend.

This PR added a env var `FLASHINFER_CUBIN_DIR` to allow people specify a cubin dir which contains pre-downloaded cubin files. When `FLASHINFER_CUBIN_DIR` is specified, we will directly return True in `has_nvidia_artifactory()`.

Differential Revision: D80035005

pavanimajety reviewed (approved):
LGTM, please rebase your branch for the CI

@mgoin for CI and further comments

houseroad reviewed (approved):
Looks good, thanks

842974287 committed:
add an env var for path to pre-downloaded flashinfer cubin files (#22675)

Summary:

Previously https://github.com/vllm-project/vllm/pull/21893 added a check for network access to NV cubin artifact endpoint. And if the check failed, we would opt out from trtllm attn backend.

This PR added a env var `FLASHINFER_CUBIN_DIR` to allow people specify a cubin dir which contains pre-downloaded cubin files. When `FLASHINFER_CUBIN_DIR` is specified, we will directly return True in `has_nvidia_artifactory()`.

Reviewed By: frank-wei, Adolfo-Karim

Differential Revision: D80035005
